src/routes/safe/components/Transactions/TxList/hooks/useTransactionType.ts

```diff
@@ -22,20 +22,24 @@ export const useTransactionType = (tx: Transaction): TxTypeProps => {
   const safeAddress = useSelector(safeAddressFromUrl)
   const toAddress = getTxTo(tx)
   // Fixed casting because known address only works for Custom tx
-  const knownAddress = useKnownAddress(toAddress?.value || '0x', {
+  const knownAddressBookAddress = useKnownAddress(toAddress?.value || '0x', {
     name: toAddress?.name || undefined,
     image: toAddress?.logoUri || undefined,
   })
 
   useEffect(() => {
     switch (tx.txInfo.type) {
       case 'Creation': {
-        setType({ icon: SettingsTxIcon, text: 'Safe created' })
+        setType({ icon: toAddress?.logoUri || SettingsTxIcon, text: 'Safe created' })
         break
       }
       case 'Transfer': {
         const isSendTx = tx.txInfo.direction === 'OUTGOING'
-        setType({ icon: isSendTx ? OutgoingTxIcon : IncomingTxIcon, text: isSendTx ? 'Send' : 'Receive' })
+
+        setType({
+          icon: isSendTx ? OutgoingTxIcon : IncomingTxIcon,
+          text: knownAddressBookAddress.name || toAddress?.name || isSendTx ? 'Send' : 'Receive',
```

Comment:
I think you might need a set of parens here
<img width="517" alt="Screenshot 2021-10-12 at 11 56 41" src="https://user-images.githubusercontent.com/381895/136934808-437b7036-9c7b-4d10-92bc-533791aaf4c1.png">
.